src/readheader.cpp

```diff
@@ -267,7 +267,14 @@ List vlrsreader(LASheader* lasheader)
     }
     else if ((strcmp(vlr.user_id, "LASF_Spec") == 0) && (vlr.data != 0))
     {
-        if (vlr.record_id == 4) // ExtraBytes
+        if (vlr.record_id == 3) // TextArea
+        {
+          CHAR* text_area_description = (CHAR*) vlr.data;
+          lvlr.push_back(nullterminate(text_area_description, vlr.record_length_after_header));
```

Comment:
The standard already imposes that the string are null terminated. No need to enforce it. The `nullterminate` function is only a trick to make the CRAN happy with (valid) non null terminated string (but we loose 1 character). We should not use it otherwise.

Reply:
Done, sorry for the iterative commit...